Return-Path: <petitboot-bounces+erichte=linux.vnet.ibm.com@lists.ozlabs.org>
Received: from g01zcilapp002.ahe.pok.ibm.com ([unix socket])
	 by g01zcilapp002 (Cyrus v2.3.11) with LMTPA;
	 Tue, 29 Oct 2019 05:26:25 -0400
X-Sieve: CMU Sieve 2.3
Received: from localhost (localhost [127.0.0.1])
	by g01zcilapp002.ahe.pok.ibm.com (Postfix) with ESMTP id 1584D2625B
	for <erichte@imap.linux.ibm.com>; Tue, 29 Oct 2019 05:26:25 -0400 (EDT)
X-Virus-Scanned: amavisd-new at linux.ibm.com
X-Spam-Flag: NO
X-Spam-Score: -1
X-Spam-Level:
X-Spam-Status: No, score=-1 tagged_above=-9999 required=6.2
	tests=[MAILING_LIST_MULTI=-1] autolearn=disabled
Received: from g01zcilapp002.ahe.pok.ibm.com ([127.0.0.1])
	by localhost (g01zcilapp002.ahe.pok.ibm.com [127.0.0.1]) (amavisd-new, port 10024)
	with LMTP id 7Iny2CnxRobw for <erichte@imap.linux.ibm.com>;
	Tue, 29 Oct 2019 05:26:24 -0400 (EDT)
Received: from g01zcilapp001.ahe.pok.ibm.com (g01zcilapp001.ahe.pok.ibm.com [9.63.16.68])
	by g01zcilapp002.ahe.pok.ibm.com (Postfix) with ESMTP id D60582625D
	for <erichte@imap.linux.ibm.com>; Tue, 29 Oct 2019 05:26:24 -0400 (EDT)
Received: from veuvn.vipa.uk.ibm.com (veuvn.vipa.uk.ibm.com [9.149.87.105])
	by g01zcilapp001.ahe.pok.ibm.com (Postfix) with ESMTP id 8913C13E002
	for <erichte@linux.ibm.com>; Tue, 29 Oct 2019 05:26:24 -0400 (EDT)
Received:  by veuvn.vipa.uk.ibm.com (IBM VM SMTP Level 640) via spool with SMTP id 2506 ; Tue, 29 Oct 2019 09:26:24 GMT
Received: by uk1vsc.vnet.ibm.com (xagsmtp 2.1.4) via smtp2 with spool id 3539;
 Tue, 29 Oct 2019 09:26:24 +0000
Received: from b03cxnp07029.gho.boulder.ibm.com [9.17.130.16] by
 veuvn.vipa.uk.ibm.com (IBM VM SMTP Level 640) via TCP with ESMTP ; Tue, 29
 Oct 2019 09:26:23 GMT
Received: from b03ledav005.gho.boulder.ibm.com
 (b03ledav005.gho.boulder.ibm.com [9.17.130.236])	by
 b03cxnp07029.gho.boulder.ibm.com (8.14.9/8.14.9/NCO v10.0) with ESMTP id
 x9T9QMMc25362844	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-GCM-SHA384
 bits=256 verify=OK)	for <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019
 09:26:22 GMT
Received: from b03ledav005.gho.boulder.ibm.com (unknown [127.0.0.1])	by IMSVA
 (Postfix) with ESMTP id D830CBE053	for <erichte@linux.vnet.ibm.com>; Tue, 29
 Oct 2019 09:26:22 +0000 (GMT)
Received: from b03ledav005.gho.boulder.ibm.com (unknown [127.0.0.1])	by IMSVA
 (Postfix) with ESMTP id 94033BE051	for <erichte@linux.vnet.ibm.com>; Tue, 29
 Oct 2019 09:26:22 +0000 (GMT)
Received: from ppma02dal.us.ibm.com (unknown [9.209.225.56])	by
 b03ledav005.gho.boulder.ibm.com (Postfix) with ESMTPS	for
 <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019 09:26:22 +0000 (GMT)
Received: from pps.filterd (ppma02dal.us.ibm.com [127.0.0.1])	by
 ppma02dal.us.ibm.com (8.16.0.27/8.16.0.27) with SMTP id x9T9Pm1F008210	for
 <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019 09:26:21 GMT
Authentication-Results: us.ibm.com;	spf=neutral
 smtp.mailfrom=petitboot-bounces+erichte=linux.vnet.ibm.com@lists.ozlabs.org;
 dkim=fail header.d=ozlabs.org header.s=201707;	dmarc=none
 header.from=ozlabs.org
Received: from mx0a-001b2d01.pphosted.com (mx0b-001b2d01.pphosted.com
 [148.163.158.5])	by ppma02dal.us.ibm.com with ESMTP id 2vvds8ge7m-1
 (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=NOT)	for
 <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019 09:26:21 +0000
Received: from pps.filterd (m0098416.ppops.net [127.0.0.1])	by
 mx0b-001b2d01.pphosted.com (8.16.0.27/8.16.0.27) with SMTP id x9T9EFls093970
 for <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019 05:26:21 -0400
Authentication-Results: ppops.net;	spf=pass
 smtp.mailfrom=petitboot-bounces+erichte=linux.vnet.ibm.com@lists.ozlabs.org;
 dkim=fail header.d=ozlabs.org header.s=201707
Received: from lists.ozlabs.org (lists.ozlabs.org [203.11.71.2])	by
 mx0b-001b2d01.pphosted.com with ESMTP id 2vxgdq4s4g-1	(version=TLSv1.2
 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=NOT)	for
 <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019 05:26:20 -0400
Received: from bilbo.ozlabs.org (lists.ozlabs.org [IPv6:2401:3900:2:1::3])	by
 lists.ozlabs.org (Postfix) with ESMTP id 472R6K2tJPzDqpq	for
 <erichte@linux.vnet.ibm.com>; Tue, 29 Oct 2019 20:26:17 +1100 (AEDT)
X-Original-To: petitboot@lists.ozlabs.org
Delivered-To: petitboot@lists.ozlabs.org
Received: from ozlabs.org (bilbo.ozlabs.org [203.11.71.1]) (using TLSv1.3 with
 cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519
 server-signature RSA-PSS (2048 bits) server-digest SHA256) (No client
 certificate requested) by lists.ozlabs.org (Postfix) with ESMTPS id
 472R132cFqzDrPt for <petitboot@lists.ozlabs.org>; Tue, 29 Oct 2019 20:21:43
 +1100 (AEDT)
Authentication-Results: lists.ozlabs.org; dmarc=pass (p=none dis=none)
 header.from=ozlabs.org
Authentication-Results: lists.ozlabs.org; dkim=pass (2048-bit key; secure)
 header.d=ozlabs.org header.i=@ozlabs.org header.b="Xr4Q3lNj";
 dkim-atps=neutral
Received: by ozlabs.org (Postfix, from userid 1023) id 472R122wCVz9sPL; Tue,
 29 Oct 2019 20:21:42 +1100 (AEDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=ozlabs.org; s=201707;
 t=1572340902; bh=EAQtNB1Jy41XWRnXsBVhKv6ugdP/apJdAkT96yVXXNw=;
 h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
 b=Xr4Q3lNjbRyZQXvdToNFhwMIKbl6jRJFhv53sXb42AbcMBza/5XS+N/3vEX0VHFMW
 MXbz/7pMM08cLT0VYoHydnwFAslorPOx1g51DXFLwL6L3U6hTxj2GhANs7eeB6s1ef
 XPPrq0bG1rkjPUIZtMvs9EHmO9cin/sBu41thYCA/hJq91D1bYX4DDU+K62pbeb6xs
 a+9rohFu8xC8nm4KVukJFIKSZDodBza2GG8mdM5tHf0Xdc/HyB9xY4ilbnlXQgFrTl
 66wCSeMtfAZHj+0ZK1vNHuX10zryYVk07j7ZJtBSB1a+++L63+H28/KsGh4vnzStyi
 b2BfSFQYTZeag==
From: Jeremy Kerr <jk@ozlabs.org>
To: petitboot@lists.ozlabs.org
Subject: [PATCH 2/2] discover/boot: unify verification failure messages
Date: Tue, 29 Oct 2019 17:15:40 +0800
Message-Id: <20191029091540.6767-2-jk@ozlabs.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20191029091540.6767-1-jk@ozlabs.org>
References: <20191029091540.6767-1-jk@ozlabs.org>
MIME-Version: 1.0
X-BeenThere: petitboot@lists.ozlabs.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: Petitboot bootloader development <petitboot.lists.ozlabs.org>
List-Unsubscribe: <https://lists.ozlabs.org/options/petitboot>,
 <mailto:petitboot-request@lists.ozlabs.org?subject=unsubscribe>
List-Archive: <http://lists.ozlabs.org/pipermail/petitboot/>
List-Post: <mailto:petitboot@lists.ozlabs.org>
List-Help: <mailto:petitboot-request@lists.ozlabs.org?subject=help>
List-Subscribe: <https://lists.ozlabs.org/listinfo/petitboot>,
 <mailto:petitboot-request@lists.ozlabs.org?subject=subscribe>
Cc: Nayna Jain <nayna@linux.ibm.com>
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
Errors-To: petitboot-bounces+erichte=linux.vnet.ibm.com@lists.ozlabs.org
Sender: 
 "Petitboot" <petitboot-bounces+erichte=linux.vnet.ibm.com@lists.ozlabs.org>
X-CLX-Shades: MLX
X-CLX-Response: 1TFkXGRgYEQpMehcbGhIRCllEF20aZhl9fRoTE0FFEQpYWBdnX0JgT1BccGR
 JUhEKeE4Xenl+ZnJAf0xlHHIRCnlMF2x+aVh6R2QdbUdTEQp5QxdhWGNzAQFbRBNyWBEKQ0gXBx
 sSGhEKQ1kXBxsSExEKWU0XZ2ZyEQpZSRcacRoQGncGGRoYcRsaHx0QGncGGBoGGhEKWV4XaGN5E
 QpJRhdDSEd1QkVZXk9OEQpDThdFEgcdfnN9REFoRFpHUl0ceE9zQH9QQ2cdUkh4GG0bRREKWFwX
 HwQaBBsSHQcTTBxLHx1JGgUbGgQbGxoEHhIEGxMTEBseGh8aEQpeWRd/HUtachEKTVwXHB0SEQp
 MWhdpaG1NTV0RCkxGF29ja2tGa2sRCkJPF2lYRVxabkNJWh0YEQpDWhcYGhkEGxsEHRsEGBEKQl
 4XGxEKREkXGxEKQkYXYn9hR0B9Tk1/Z0sRCkJHF25GRnx5HRl+GVsfEQpCXBcaEQpCRRdocAFbQ
 EISH2gfTxEKQk4Xenl+ZnJAf0xlHHIRCkJMF2dfQmBPUFxwZElSEQpCbBdoHX1EYFhvGHxsZxEK
 QkAXZmAYU3NDR1tcREIRCkJYF2IeaW1lYB9HYB8SEQpaWBcYEQpwZxdmEmFSYWl4AX58RBAaEQp
 waBdvclNfHU1zX0Z7TxAaEQpwaBdraGhCeXIfXlITWhAaEQpwaBdocE55UlxNYQEeeRAaEQpwaB
 dkfF5EGk5kTn9LHBAaEQpwaBdsRHJrbVNmGFtocxAaEQpwfxdmRk1OZVJEElBMYxAbGhoRCnBfF
 3paZx9NBWRLRFBQEBoRCnB9F216UHNCbh5jaxxMEBoRCnBsF2NtZX58E2lzGEYfEB4SEQptfhca
 EQpYTRdLESA=
X-Proofpoint-Virus-Version: vendor=fsecure engine=2.50.10434:,,
 definitions=2019-10-29_03:,, signatures=0
X-Proofpoint-Spam-Details: rule=inbound_notspam policy=inbound score=0
 priorityscore=108 malwarescore=0 suspectscore=0 phishscore=0 bulkscore=0
 spamscore=0 clxscore=322 lowpriorityscore=0 mlxscore=0 impostorscore=0
 mlxlogscore=999 adultscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.0.1-1908290000 definitions=main-1910290096
X-TM-AS-GCONF: 00
X-Xagent-Gateway: uk1vsc.vnet.ibm.com (XAGSMTP2 at UK1VSC)

Currently, we have two sites where the result of validate_boot_files is
interpreted: in kexec_load, and boot_process. In the former, we generate
the pb_log message, and in the latter we generate the status message.

This means we have separate places to maintain similar error messages,
which is prone to future errors. This change does all of the
interpretation directly after calling validate_boot_files().

Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
---
 discover/boot.c | 45 ++++++++++++++++++++++-----------------------
 1 file changed, 22 insertions(+), 23 deletions(-)

diff --git a/discover/boot.c b/discover/boot.c
index a6b88f0..9e7054b 100644
--- a/discover/boot.c
+++ b/discover/boot.c
@@ -75,16 +75,30 @@ static int kexec_load(struct boot_task *boot_task)
 	boot_task->local_dtb_override = NULL;
 	boot_task->local_image_override = NULL;

-	if ((result = validate_boot_files(boot_task))) {
-		if (result == KEXEC_LOAD_DECRYPTION_FALURE) {
-			pb_log("%s: Aborting kexec due to"
-				" decryption failure\n", __func__);
-		}
-		if (result == KEXEC_LOAD_SIGNATURE_FAILURE) {
-			pb_log("%s: Aborting kexec due to signature"
-				" verification failure\n", __func__);
+	result = validate_boot_files(boot_task);
+	if (result) {
+		const char *msg;
+
+		switch (result) {
+		case KEXEC_LOAD_DECRYPTION_FALURE:
+			msg = _("decryption failed");
+			break;
+		case KEXEC_LOAD_SIGNATURE_FAILURE:
+			msg = _("signature verification failed");
+			break;
+		case KEXEC_LOAD_SIG_SETUP_INVALID:
+			msg = _("invalid signature configuration");
+			break;
+		default:
+			msg = _("unknown verification failure");
 		}

+		update_status(boot_task->status_fn, boot_task->status_arg,
+				STATUS_ERROR,
+				_("Boot verification failure: %s"), msg);
+		pb_log_fn("Aborting kexec due to verification failure: %s",
+				msg);
+
 		validate_boot_files_cleanup(boot_task);
 		return result;
 	}
@@ -451,21 +465,6 @@ static void boot_process(struct load_url_result *result, void *data)
 			_("Performing kexec load"));

 	rc = kexec_load(task);
-	pb_log_fn("kexec_load returned %d\n", rc);
-	if (rc == KEXEC_LOAD_DECRYPTION_FALURE) {
-		update_status(task->status_fn, task->status_arg,
-				STATUS_ERROR, _("Decryption failed"));
-	}
-	else if (rc == KEXEC_LOAD_SIGNATURE_FAILURE) {
-		update_status(task->status_fn, task->status_arg,
-				STATUS_ERROR,
-				_("Signature verification failed"));
-	}
-	else if (rc == KEXEC_LOAD_SIG_SETUP_INVALID) {
-		update_status(task->status_fn, task->status_arg,
-				STATUS_ERROR,
-				_("Invalid signature configuration"));
-	}

 no_load:
 	list_for_each_entry(&task->resources, resource, list)
-- 
2.20.1

_______________________________________________
Petitboot mailing list
Petitboot@lists.ozlabs.org
https://lists.ozlabs.org/listinfo/petitboot
